pkg/provider/acme/provider.go

```diff
@@ -468,6 +500,54 @@ func (p *Provider) watchNewDomains(ctx context.Context) {
 	})
 }
 
+func (p *Provider) resolveDefaultCertificate(ctx context.Context, domain types.Domain, tlsStore string) (*certificate.Resource, error) {
+	logger := log.FromContext(ctx)
+
+	domains, err := p.getValidDomains(ctx, domain)
+	if err != nil {
+		return nil, err
+	}
+
+	// Continue only if no domains of the default cert are being resolved at the moment.
+	if p.containsResolvingDomains(domains) {
+		return nil, nil
+	}
+
+	p.addResolvingDomains(domains)
+	defer p.removeResolvingDomains(domains)
+
+	logger.Debugf("Loading ACME certificates %+v...", domains)
+
+	client, err := p.getClient()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get ACME client %w", err)
+	}
+
+	request := certificate.ObtainRequest{
+		Domains:        domains,
+		Bundle:         true,
+		MustStaple:     oscpMustStaple,
```

Comment:
Shouldn't this be`ocspMustStaple` (for [OCSP](https://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol))?